### PR TITLE
server api

### PR DESCRIPTION
**Pull request by emkwak**

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -2,9 +2,11 @@ const express = require('express');
 const app = express();
 const port = 3001;
 const db = require('../database/index.js')
+const rsvpRouter = require('./routes/rsvp');
 
 app.use(express.static('public'));
 
+app.use('/rsvp', rsvpRouter);
```
Comment (manuelag19) [inline server/index.js]: Ah a router, now I get what you mean by a separate place for the routes.

File: server/routes/rsvp.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
+const router = require('express').Router();
+let Event = require('../../database/models/Event.js');
+
+router.route('/').get((req, res) => {
+  Event.find()
+    .then(events => res.json(events))
```
Comment (manuelag19) [inline server/routes/rsvp.js]: Are you using this endpoint where you return all the events?

File: server/routes/rsvp.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    .then((event) => {
+      var eventHosts = []
+      event.eventOrganizer.forEach(user => {
+        const person = {
+          name: user.name,
+          thumbnail: user.thumbnail
+        }
+        eventHosts.push(person)
+      })
+      res.json({ eventHosts })
```
Comment (manuelag19) [inline server/routes/rsvp.js]: This means the endpoint /rsvp/hosts/:eventId returns {eventHosts: [...]}. The app service plan only returns the array.

File: server/routes/rsvp.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    })
+    .catch(err => res.status(400).json(`Error: ${err}`));
+})
+
+router.route('/profiles/:eventId').get((req, res) => {
+  Event.findOne({ eventId: req.params.eventId })
+    .then((event) => {
+      var attendees = []
```
Comment (manuelag19) [inline server/routes/rsvp.js]: Might be better named profiles to differentiate from attendees below.

File: server/routes/rsvp.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        const person = {
+          memberId: user.memberId,
+          name: user.name,
+          thumbnail: user.thumbnail
+        }
+        attendees.push(person)
+      })
+      res.json({ attendees })
+    })
+    .catch(err => res.status(400).json(`Error: ${err}`));
```
Comment (manuelag19) [inline server/routes/rsvp.js]: I was wondering if 400 was the right error and I found this: https://stackoverflow.com/questions/25378624/should-a-restful-api-return-400-or-404-when-passed-an-invalid-id
You’re not validating whether the event id is a number so I’m not sure which error status would be appropriate.

File: server/routes/rsvp.js (hunk 2 above)
Comment (emkwak) [reply server/routes/rsvp.js]: @manuelag19 I removed this

File: server/routes/rsvp.js (hunk 4 above)
Comment (emkwak) [reply server/routes/rsvp.js]: @manuelag19 I also removed this. This was for Will's microservice

**Comment by manuelag19** [comment]
Looks good. Yeah, I forgot we no longer need to serve endpoints for Will’s microservice.

**Pull request merged**